src/veins/modules/obstacle/ObstacleControl.cc

```diff
@@ -179,6 +189,297 @@ void ObstacleControl::erase(const Obstacle* obstacle) {
+
+	if (hasGUI() && annotations) {
+		annotations->eraseAll(vehicleAnnotationGroup);
+		drawVehicleObstacles(sStart);
+		annotations->drawLine(senderPos, receiverPos, "blue", vehicleAnnotationGroup);
+	}
+
+	double x1 = std::min(senderPos.x, receiverPos.x);
+	double x2 = std::max(senderPos.x, receiverPos.x);
+	double y1 = std::min(senderPos.y, receiverPos.y);
+	double y2 = std::max(senderPos.y, receiverPos.y);
+
+	for (VehicleObstacles::const_iterator i = vehicleObstacles.begin(); i != vehicleObstacles.end(); ++i) {
+		const TraCIMobility* m = (*i)->getTraCIMobility();
+		Coord p  = m->getCurrentPosition();//PositionAt(sStart);
+		double l = (*i)->getLength();
+		double w = (*i)->getWidth();
+		double h = (*i)->getHeight();
+
+		EV << "checking vehicle at " << p.info() << " with height: " << h << " width: " << w << " length: " << l << endl;
+		// shortcut if AABBs can't overlap
+		double lw = std::max(l, w);
+		if ((p.x + lw) < x1) continue;
+		if ((p.x - lw) > x2) continue;
+		if ((p.y + lw) < y1) continue;
+		if ((p.y - lw) > y2) continue;
+
+		if (p == senderPos) {
+			// this is the sender
+			senderHeight = h;
+		}
+		else if (p == receiverPos) {
+			// this is the receiver
+			receiverHeight = h;
+		}
+		else {
+			// this is a potential obstacle
+			double p1d = (*i)->getIntersectionPoint(senderPos, receiverPos);
+			double maxd = senderPos.distance(receiverPos);
+			if (!std::isnan(p1d) && p1d > 0 && p1d < maxd) {
+				std::vector<std::pair<double, double> >::iterator it = potentialObstacles.begin();
+				while (1) {
+					if (it == potentialObstacles.end()) {
+						potentialObstacles.push_back(std::make_pair(p1d, h));
+						break;
+					}
+					if (it->first == p1d) { //omit double entries
+						EV << "two obstacles at same distance " << it->first << " == " << p1d << " height: " << it->second << " =? " << h << std::endl;
+						break;
+					}
+					if (it->first > p1d) {
+						potentialObstacles.insert(it, std::make_pair(p1d, h));
+						break;
+					}
+					++it;
+				}
+				EV << "\tgot obstacle in 2d-LOS at " << p.info() << ", " << p1d << " meters away from sender" << std::endl;
+				Coord hitPos = senderPos + (receiverPos - senderPos) / senderPos.distance(receiverPos) * p1d;
+				if (hasGUI() && annotations)
+					annotations->drawLine(senderPos, hitPos, "red", vehicleAnnotationGroup);
+			}
+		}
+
+	}
+	EV << "senderHeight: " << senderHeight << endl;
+	EV << "receiverHeight: " << receiverHeight << endl;
+	ASSERT(!std::isnan(senderHeight));
+	ASSERT(!std::isnan(receiverHeight));
+	if (std::isnan(senderHeight)) return 1;
+	if (std::isnan(receiverHeight)) return 1;
+	if (potentialObstacles.size() < 1) return 1;
+
+	potentialObstacles.insert(potentialObstacles.begin(), std::make_pair(0, senderHeight));
+	potentialObstacles.push_back(std::make_pair(senderPos.distance(receiverPos), receiverHeight));
+
+	double attenuationDB = getVehicleAttenuation(potentialObstacles, 5.9);  // FIXME: 5.9 is hardcoded
```

Comment:
this number can probably be pulled directly out of the `omnetpp.ini` config?

Reply:
[implemented here](https://github.com/sommer/veins/pull/35/commits/efc8aa7ad23fdf197dc5c78edcf95489f787f524)